### PR TITLE
grub2: workaround parallel build failure by building .po files sequentially

### PR DESCRIPTION
**Pull request by trofi**
…tially

Without the change `grub2` build occasionally fails as:

    https://hydra.nixos.org/log/h04imslmgy9l33z6xxsaf9v4svqsc3n2-grub-2.12-rc1.drv

    $ msgmerge --update  --lang=de de.po grub.pot
    $ msgmerge --update  --lang=de de.po grub.pot
    ...
    de.po:745: end-of-file within string
    de.po:745: missing 'msgstr' section

Here `msgmerge --update` attempts to update `de.po` from two parallel `make` calls when translation updates are happening.

The change works around `de.po` stomping bu regenerating translations sequentially.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by trofi** [comment]
Result of `nixpkgs-review pr 248886` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>11 packages built:</summary>
  <ul>
    <li>grub2 (grub2_light)</li>
    <li>grub2.debug (grub2_light.debug)</li>
    <li>grub2_efi</li>
    <li>grub2_efi.debug</li>
    <li>grub2_pvgrub_image</li>
    <li>grub2_xen</li>
    <li>os-prober</li>
    <li>timeshift</li>
    <li>woeusb</li>
    <li>woeusb-ng</li>
    <li>woeusb-ng.dist</li>
  </ul>
</details>

**Comment by tpwrules** [comment]
I was able to get six consecutive builds to work without failing. Seems that this solves the issue.

Thanks very much for quick response and a proper fix!

I wonder if the comment should be changed to link to this PR instead of mine?

**Comment by trofi** [comment]
> I was able to get six consecutive builds to work without failing. Seems that this solves the issue.

Aha, nice! Apparently I forgot my loop-test in the background and it survived 120 `grub2` rebuilds. Previously it used to fail on about every ~10th run.

> I wonder if the comment should be changed to link to this PR instead of mine?

I linked the PR as a source of a bit of debugging (it's not too much details here). I linked specific comment instead to make it a bit clearer and re-pushed:

```diff
--- a/pkgs/tools/misc/grub/default.nix
+++ b/pkgs/tools/misc/grub/default.nix
@@ -151,7 +151,7 @@ stdenv.mkDerivation rec {
   postConfigure = ''
     # make sure .po files are up to date to workaround
     # parallel `msgmerge --update` on autogenerated .po files:
-    #   https://github.com/NixOS/nixpkgs/pull/248747
+    #   https://github.com/NixOS/nixpkgs/pull/248747#issuecomment-1676301670
     make dist
   '';
```

**Comment by tpwrules** [comment]
Ah right the original thread did have more debugging. And for whatever reason on my system it fails maybe 70% of the time. Way more frequently than 10%. As a result I'd like to see this merged ASAP.

File: pkgs/tools/misc/grub/default.nix [hunk 1]
```diff
@@ -148,6 +148,12 @@ stdenv.mkDerivation rec {
 
       substituteInPlace ./configure --replace '/usr/share/fonts/unifont' '${unifont}/share/fonts'
     '';
+  postConfigure = ''
```
Comment (samueldr) [inline pkgs/tools/misc/grub/default.nix:151]:
```suggestion

  postConfigure = ''
```

**Comment by samueldr** [comment]
I haven't tested, but it smells right from reading it. So assuming it works, it's approved for all but the trivial nit.

**Comment by trofi** [comment]
Squashed/rebased with a newline added after `postConfigure`.

**Pull request merged**